lightningd/lightningd.c

```diff
@@ -133,6 +135,28 @@ static void test_daemons(const struct lightningd *ld)
 	}
 	tal_free(ctx);
 }
+/* Check if all the daemons exist in the specified directory. */
+static bool has_all_daemons(const char* daemon_dir)
+{
+	size_t i;
+	bool retval;
+	const tal_t *ctx = tal_tmpctx(NULL);
+
+	/* If we complete the below loop, all daemons could be found
+	 * as files in the directory. */
+	retval = true;
+	for (i = 0; i < ARRAY_SIZE(daemons); ++i) {
+		const char *dpath = path_join(ctx, daemon_dir, daemons[i]);
+		if (!path_is_file(dpath)) {
+			retval = false;
+			break;
+		}
+		tal_free(dpath);
```

Comment:
Don't need this tal_free, just rely on cleanup at the bottom.  Mildly more efficient, too.

I'd probably have coded this as naively as possible, ie:

```
for (size_t i = 0; i < ARRAY_SIZE(daemons); ++i) {
   if (!path_is_file(path_join(tmpctx, daemon_dir, daemons[i]))
      daemon_missing = true;
}
tal_free(tmpctx);
return !daemon_missing;
```

Reply:
Sorry, not yet that familiar with the `tal` system, will modify.